BedrockCore.cpp

```diff
@@ -30,29 +30,39 @@ class AutoScopeRewrite {
 };
 
 uint64_t BedrockCore::_getTimeout(const SData& request) {
-    uint64_t timeout = request.isSet("timeout") ? request.calc("timeout") : DEFAULT_TIMEOUT;
+
+    // Timeout is the default, unless explicitly supplied, or if Connection: forget is set.
+    uint64_t timeout =  DEFAULT_TIMEOUT;
+    if (request.isSet("timeout")) {
+        timeout = request.calc("timeout");
+    } else if (SIEquals(request["connection"], "forget")) {
+        timeout = DEFAULT_TIMEOUT_FORGET;
+    }
+
+    // We also want to know the processTimeout, because we'll return early if we get stuck processing for too long.
+    int64_t processTimeout = request.isSet("processTimeout") ? request.calc("processTimeout") : DEFAULT_PROCESS_TIMEOUT;
 
     // See when the command was scheduled to run. The timeout is from *this* start time, not from when the command
     // starts executing.
     try {
         int64_t adjustedTimeout = (int64_t)timeout - (int64_t)((STimeNow() - stoull(request["commandExecuteTime"])) / 1000);
 
         // If this is negative, we're *already* past the timeout, just return early.
-        if (adjustedTimeout <= 0) {
+        if (adjustedTimeout <= 0 || processTimeout <= 0) {
```

Comment:
how is `processsTimeout` ever going to be `<= 0`? The only place it is set is L43, where it's set to whatever the request has or the default then it's never changed. Don't you need to also computer an `adjustedProcessTimeout` like we do on L48?

Reply:
If someone sends:
```
SomeCommand
processTimeout: -1

```

It's stored in a signed number so that we can compare it to `adjustedTimeout` (which is expected to be negative, sometimes) so there's nothing strictly preventing it from being negative.

Yeah, don't do that, but still.

> Don't you need to also computer an adjustedProcessTimeout like we do on L48?

Nope, process in per `peek` or `process` invocation. As long as no single instance of `peek` or `process` takes longer than `processTimeout`, the command won't stop executing because of this. If `processTimeout` is 10s we can return 10s from `_getTimeout` repeatedly, in case it gets called multiple times, and that's fine eventually the regular `timeout` will elapse and we'll return that instead, when it's less than 10s.